### PR TITLE
Allow using tile metadata to store a tighter bounding box for implicit tiling

### PR DESCRIPTION
**Pull request by ptrgags**
This PR + bit of metadata in the tileset allows `viewer.zoomTo()` to zoom to a tileset correctly when the root bounding box is loose.

My team's been working with implicit tiling octrees lately. For datasets that are wide but not very tall, we want to prevent super thin bounding volumes (which can be bad for culling), so we make the root bounding volume a cube instead, like this:

image

The problem is `viewer.zoomTo()` zooms to the bounding volume's bounding sphere (which is centered on the _center_ of the tile, which for the type of tileset above puts the camera way too high. Compare these screenshots/Sandcastles:

| main ([hosted Sandcastle](https://sandcastle.cesium.com/#c=bVNrb9owFP0rV3wpSCyBtTC60WodUJaoAbVNH6BIk0kuYHBsZptHmPrfd0No10375Nj3nONzz3VcF/qaSQsdNHyd+vfA4hiNAasgU2sNXElgxqA1kSwwjqekk+CUrYW9OoBDtUQJFxCVMPPnk37Mh9z3HvZefcA948m7Rtzxmt5y9fzY8c8dAv1M+ksCebVx+LgIFr3dKKttR0+P4ia8Swf7pR12g7MBr+0H+zsRLJLFuOvtxmnAbzr+akxiQThrBOGSxAeLyek3EWde84kuj0+9HCNHmZ/k2NHzLR8uevXB/vbjsDurB3sywHBUa/p2u9yMeCPIRO9H7Cet2w16u83iwZt86J19t32/OZvMotKXSEYyVtJY2HDcoqZGJW6PgTmPh7NyVIoP+46SlnGJOipVDkyrM/gVSYBCwnKBFCZpsC3jr7E7xXLaDYuyM9UqpZiv8uC9pFxvfWo1WufVQgnAdcM5N29iiUIjTyzM2QaBgVAxszS3KhgFWzzRCGvD5YxKqUpQBMxqvssnvBIsRiAbzEKtCjXg5BFZAmoKiWbbA4lMzxFilJZ6p0K+Q6btvPDyTvLzazshPSkzVTo1DjJjB4rQD6tQXfMdJteapVguyPAWAAnSF5Onh967ONOIppybqhTQSjVfX/JU4TgJx5ArdFaap9zyDRqHJUn5mEoBLFI+wvdKpaF6D3gBiiqeQxm1VrryZ1JKoCPU7HieIyNZqpbaxmYCL3PUV56uqC9Ya1F2HNdiSmlSD+5kHS9phLExOa/tvlLaCd8ATy7+81QgFvSPUWW6FuKe7zEqXbZdwv9FE4olNJHhBrVgGUFyG+15/fKmKDiO03Zpm1/6L9cqJSZMv9P9DQ)) | this branch ([local Sandcastle](http://localhost:8080/Apps/Sandcastle/index.html#c=bZNtb9owFIX/yhVfBhJzYBRGN1qtA8oSNaC2KS0o0mSSCxgcm9nmJZn63+cQ2lVbPyW2n3ty7rmO48BAUWGgi5ptE+8eaBSh1mAkpHKrgEkBVGs0OhQFQ1wpSIxzuuXm6ggHco0CLiAsYeotZ4OIjZjnPmRufchc7Yq7ZtR1W+568zTueufEQr/iwdpCbm0ajFf+qn+YpLX95HHMb4K7ZJitzajnnw1ZLRtmd9xfxatpzz1ME5/ddL3N1Ir5waLpB2srPlzNGt95lLqtR/vxqOHmjJikXpyzk6dbNlr168Ps9tOot6j7mTVAcVJreWa/3k1Y0095/2fkxe3bHbqH3erBnX3sn/0wA6+1mC3C0tdQhCKSQhvYMdyjso0K3J8CI+PjXjksRcd1VwpDmUAVlirHSqNS+B0KgELCMI42TKtB95S9xE6KR6MXFMdkrmRiY77Kg3fjcr39ud1sn1cLJQDHCZZMv4rFErX4YGBJdwgUuIyosXOrgpawxw8KYauZWNijRMbIfWoUO+QT3nAaIVgb1ECtCjVg1iPSGOQcYkX3xyJreokQoTC2d3uQr5Aqsyy8vJH88tJOYK+UnkuVaIJUm6G09MMmkNfsgPG1ogmWi2J4DcAK2jcqGsfee7hQiLqcm6oUaKWaP5/zVOE0CaKtKyQbxRJm2A41oXFcPqVSgEXKJzyTMgnkW+AZbFTREsqolFSVv5OSHAmXi9N+ToaiVC11tEk5Xr6Y/8aSje0NtoqXCXEMJjZR24cz20ZrO8ZI68IGQMd5W9qJ2Q5YfPHOtYGI2//Nnsy3nN+zDMPSZcex/H+lXNLYTmi0Q8VpmmPL+uVNsUkI6Th2+X6lkZLPqPpH+Q8)) |
| ---- | ------- | 
| image | image |

What's the fix?

* the tileset has extra metadata, `TILE_BOUNDING_BOX`, here used as a tighter bounding box than the root bounding box
```jsonc
{
  "schema": {
    "id": "cesium-tiling-pipeline",
    "classes": {
      "tile": {
        "properties": {
          "tightBoundingBox": {
            "name": "Tight Bounding Box",
            "type": "SCALAR",
            "componentType": "FLOAT64",
            "array": true,
            "count": 12,
            // This semantic is what this PR detects
            "semantic": "TILE_BOUNDING_BOX"
          }
        }
      }
    }
  },
  "root": {
    // Loose bounding volume, used for implicit tiling
    "boundingVolume": {
      "box": [33.69379425048828, 18.022323608398438, 229.1623935699463, 281.7328109741211, 0.0, 0.0, 0.0, 281.73280334472656, 0.0, 0.0, 0.0, 281.73280906677246]
    },
    // ...
    "metadata": {
      "class": "tile",
      "properties": {
        // tight bounding volume, this overrides the bounding volume for the "placeholder tile" for implicit tiling so the camera zooms to the tight bounds instead of the loose ones.
        "tightBoundingBox": [16.432838500000003, 18.02232349999997, -32.271462, 264.4718555, 0.0, 0.0, 0.0, 281.7328035, 0.0, 0.0, 0.0, 20.298953000000004]
      }
    },
    "implicitTiling": {
      //...
    }
  }
}
```
* This PR modifies `Cesium3DTile` to parse 3D Tiles 1.1 [metadata semantics](https://github.com/CesiumGS/3d-tiles/tree/draft-1.1/specification/Metadata/Semantics#overview-4) `TILE_BOUNDING_(BOX|REGION|SPHERE)` and use this instead of `tile.boundingVolume` when constructing the bounding volume for rendering purposes.

Since this metadata is only listed on the "placeholder tile" for implicit tiling, it does not propagate to the tiles generated by `Implicit3DTileContent`. This is intentional, as only the root tile affects `viewer.zoomTo()`. 

@jjhembd could you review?

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @ptrgags!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium/blob/tile-bv-metadata/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.
* :grey_question: Unit tests were not updated.
    * Make sure you've [updated tests](https://github.com/CesiumGS/cesium/tree/main/Documentation/Contributors/TestingGuide) to reflect your changes, added tests for any new code, and ran the code coverage tool.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

File: packages/engine/Specs/Scene/BoundingVolumeSemanticsSpec.js [hunk 1]
```diff
@@ -0,0 +1,437 @@
+import {
+  TileMetadata,
+  Math,
+  MetadataClass,
+  BoundingVolumeSemantics,
+} from "../../index.js";
+
+describe("Scene/parseBoundingVolumeSemantics", function () {
```
Comment (jjhembd) [inline packages/engine/Specs/Scene/BoundingVolumeSemanticsSpec.js:8]: I think this spec should be renamed to `BoundingVolumeSemantics`

File: packages/engine/Source/Scene/BoundingVolumeSemantics.js [hunk 2]
```diff
@@ -98,10 +138,18 @@ function parseBoundingVolume(prefix, tileMetadata) {
  * @return {number} The minimum height
  * @private
  */
-function parseMinimumHeight(prefix, tileMetadata) {
+BoundingVolumeSemantics.parseMinimumHeight = function (prefix, tileMetadata) {
```
Comment (jjhembd) [inline packages/engine/Source/Scene/BoundingVolumeSemantics.js:141]: This method is not used outside the class. Can we prefix the name with an underscore to make the intended scope more clear?

File: packages/engine/Source/Scene/BoundingVolumeSemantics.js [hunk 3]
```diff
@@ -113,9 +161,17 @@ function parseMinimumHeight(prefix, tileMetadata) {
  * @return {number} The maximum height
  * @private
  */
-function parseMaximumHeight(prefix, tileMetadata) {
+BoundingVolumeSemantics.parseMaximumHeight = function (prefix, tileMetadata) {
```
Comment (jjhembd) [inline packages/engine/Source/Scene/BoundingVolumeSemantics.js:164]: This internal method could also be underscored

**Review by jjhembd — COMMENTED** [review]
Thanks @ptrgags! I think this looks good overall, I just have a couple minor comments about naming.

I like the improved testability of the new structure in `BoundingVolumeSemantics`!

**Comment by jjhembd** [comment]
Thanks again @ptrgags!

**Pull request merged**